aztk/spark/client.py

```diff
@@ -131,3 +136,124 @@ def get_application_status(self, cluster_id: str, app_name: str):
             return task.state._value_
         except batch_error.BatchErrorException as e:
             raise error.AztkError(helpers.format_batch_exception(e))
+    
+    '''
+        job submission
+    '''
+    def submit_job(self, job_configuration):
+        try:
+            zip_resource_files = upload_node_scripts.zip_scripts(self.blob_client,
+                                                                 job_configuration.id,
+                                                                 job_configuration.custom_scripts,
+                                                                 job_configuration.spark_configuration)
+            
+            start_task = create_cluster_helper.generate_cluster_start_task(self,
+                                                                           zip_resource_files,
+                                                                           job_configuration.gpu_enabled,
+                                                                           job_configuration.docker_repo)
+             
+            application_tasks = []
+            for application in job_configuration.applications:
+                application_tasks.append((application, submit_helper.generate_task(self, job_configuration.id, application)))
+
+            job_manager_task = job_submit_helper.generate_task(self, job_configuration, application_tasks)
+
+
+            software_metadata_key = "spark"
+
+            vm_image = models.VmImage(
+                publisher='Canonical',
+                offer='UbuntuServer',
+                sku='16.04')
+            
+            if job_configuration.max_dedicated_nodes and not job_configuration.max_low_pri_nodes:
```

Comment:
Do you need this logic? Can't you always just set them both but one will always be 0?

Reply:
For the CLI, I have validation in place for when the file is read. However, for the SDK, I'm not sure where the best place to put the validation is. I agree that this method probably shouldn't have logic to validate the job_configuration objection. But, I'm not sure where the best place to validate is. Perhaps on the model initialization. What do you think?